cmd/kubeadm/app/cmd/init.go

```diff
@@ -150,6 +150,11 @@ func newCmdInit(out io.Writer, initOptions *initOptions) *cobra.Command {
 	// sets the data builder function, that will be used by the runner
 	// both when running the entire workflow or single phases
 	initRunner.SetDataInitializer(func(cmd *cobra.Command, args []string) (workflow.RunData, error) {
+		if cmd.Flags().Lookup(options.NodeCRISocket) == nil {
+			// avoid CRI detection
+			// assume that the command execution does not depend on CRISocket when --cri-socket flag is not set
+			initOptions.externalInitCfg.NodeRegistration.CRISocket = kubeadmconstants.UnknownCRISocket
```

Comment:
yet, doesn't this mean that socket detection will never happen?
if the flag is not passed and the socket is not set on config we should do the dynamic defaulting / detection.

Reply:
i wonder what is best here...perhaps phases that do need the socket can explicitly remove the unknown socket setting and do the detection on demand.

Reply:
The socket detection is performed when the user executes a command that requires socket detection, that is, a command that contains `--cri-socket` flag. In general, when a command relies on socket detection, we need to ensure that it contains the `--cri-socket` flag to ensure consistency in the experience.

Reply:
I checked to make sure all the commands were correctly configured with the `--cri-socket` flag. 
But hope for double check 😄

Reply:
hm, wasn't socket detection also done when --config is passed. or did we change this behavior?

Reply:
When `--config` is passed:
- If `--cri-socket` is not passed, whether to check the socket should depend on whether the user specified `nodeRegistration.criSocket` in the configuration file. 
- If `--cri-socket` is also passed, will just use it, and the detection will be skipped.

I don't think it's going to change the original behavior.

Reply:
Do we want to install the `--cri-socket` flag on and only on those commands that require CRI info?
If yes, I think this doesn't change the original behavior, and it can skip socket detection when it's not needed.

Reply:
> Do we want to install the --cri-socket flag on and only on those commands that require CRI info

yes